crmd/lrm.c

```diff
@@ -2331,9 +2331,18 @@ process_lrm_event(lrm_state_t * lrm_state, lrmd_event_data_t * op, struct recurr
     CRM_CHECK(op != NULL, return FALSE);
     CRM_CHECK(op->rsc_id != NULL, return FALSE);
 
+    rsc = lrm_state_get_rsc_info(lrm_state, op->rsc_id, 0);
+
+    if (op->op_status == PCMK_LRM_OP_DONE && op->rc == PCMK_OCF_OK &&
+        safe_str_eq(op->op_type, CRMD_ACTION_METADATA) && rsc) {
+
+        lrm_state_update_rsc_metadata(lrm_state, rsc, op->output);
```

Comment:
Since now metadata update is initiated by tengine only, not by passing NULL to lrm_state_update_rsc_metadata() we're out of danger.